lib/grp.gi

```diff
@@ -1342,6 +1342,50 @@ end);
 ##
 #M  MaximalSubgroups( <G> )
 ##
+InstallMethod(MaximalSubgroupClassReps,"default, catch dangerous options",
+  true,[IsGroup],0,
+function(G)
+local H,a,m,i,l;
+  # easy case, go without options
+  if not HasTryMaximalSubgroupClassReps(G) then
+    return TryMaximalSubgroupClassReps(G:
+           # as if options were unset
+           cheap:=fail,intersize:=fail,inmax:=fail,nolattice:=fail);
+  fi;
+
+  # hard case -- `Try` is stored
+  if not IsBound(G!.maxsubtrytaint) or G!.maxsubtrytaint=false then
+    # stored and untainted, just go on
+    return TryMaximalSubgroupClassReps(G); 
+  fi;
+
+  # compute anew for new group to avoid taint
+  H:=Group(GeneratorsOfGroup(G));
+  for i in [Size,IsNaturalAlternatingGroup,IsNaturalSymmetricGroup] do
+    if Tester(i)(G) then Setter(i)(H,i(G));fi;
+  od;
+  m:=TryMaximalSubgroupClassReps(H:
+          cheap:=false,intersize:=false,inmax:=false,nolattice:=false);
+  l:=[];
+  for i in m do
+    a:=SubgroupNC(G,GeneratorsOfGroup(i));
+    if HasSize(i) then SetSize(a,Size(i));fi;
+    Add(l,a);
+  od;
+
+  # now we know list is untained, store 
+  Unbind(G!.maxsubtrytaint);
+  G!.TryMaximalSubgroupClassReps:=l;
+  return l;
+
+end);
+
+InstallGlobalFunction(TryMaxSubgroupTainter,function(G)
```

Comment:
Things get even simpler if you also add an operation `TryMaximalSubgroupClassRepsOp`, and move most current `MaximalSubgroupClassReps` resp. `TryMaximalSubgroupClassReps` methods to that operation. Then, both `MaximalSubgroupClassReps` resp. `TryMaximalSubgroupClassReps` can dispatch to that operation. This would also remove the need to create a temporary "copy" of the group, and later transfer subgroups back to the original group.

And the main (only?) `TryMaximalSubgroupClassReps` method can look roughly like this:
```
function(G)
  local maxs;
  if HasMaximalSubgroupClassReps(G) then return MaximalSubgroupClassReps(G); fi;
  maxs := TryMaximalSubgroupClassRepsOp(G);
  if NO-BAD-OPTIONS-SET then
    SetMaximalSubgroupClassReps(G, maxs);
  fi;
  return maxs`
end;
```
That's still slightly brittle, because some `TryMaximalSubgroupClassRepsOp` might accept more options in the future, and if we forget to update the above method, we have a problem... But at least we won't be off worse than we are now.

(BTW, I'd actually turn those options into explicit arguments to `TryMaximalSubgroupClassRepsOp`, but that's probably a matter of taste)

Reply:
PS: of course one could also try to come up with a better name... perhaps the second attribute should be `SomeMaximalSubgroupClassReps`, and the operation then could be `TryMaximalSubgroupClassReps`, or `ComputeSomeMaximalSubgroupClassReps`, or ... but that's of course in the end irrelevant, the functionality is what counts.